frontend/src/pages/user/UserAnswerPage.js

```diff
@@ -1,33 +1,128 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getUserData } from "../../api/api";
 
 export default function UserAnswerPage() {
+  const [title, setTitle] = useState("");
+  const [questions, setQuestions] = useState([]);
+  const [score, setScore] = useState(0);
+  const [answers, setAnswers] = useState([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [questionsPerPage] = useState(1);
+  const { categoryId } = useParams();
+
+  useEffect(() => {
+    const fetch = async () => {
+      const response = await getUserData(categoryId);
+      setTitle(response.title);
+      setQuestions(response.questions);
```

Comment:
You can set an object to a state. No need to declare 1 by 1
```suggestion
      setQuestions(response);
```